ui/src/app/stores/workflows/workflows.reducers.ts

```diff
@@ -16,27 +16,141 @@
+          ...initialState.workflowAction, loading: false
+        }};
+    case (WorkflowsActions.LOAD_WORKFLOW_FAILURE_INCORRECT_ID):
+      return {...state, workflowAction: {
+          ...initialState.workflowAction, loading: false
+        }};
+    case (WorkflowsActions.WORKFLOW_DETAILS_CHANGED):
+      const detailsData = [
+        ...state.workflowAction.workflowData.details.filter(item => item.property !== action.payload.property),
+        action.payload
+      ];
+      return {...state, workflowAction: {
+          ...state.workflowAction, workflowData: {
+            ...state.workflowAction.workflowData, details: [...detailsData]
+          }
+        }};
+    case (WorkflowsActions.WORKFLOW_SENSOR_CHANGED):
+      const sensorData = [
+        ...state.workflowAction.workflowData.sensor.filter(item => item.property !== action.payload.property),
+        action.payload
+      ];
+      return {...state, workflowAction: {
+          ...state.workflowAction, workflowData: {
+            ...state.workflowAction.workflowData, sensor: [...sensorData]
+          }
+        }};
+    case (WorkflowsActions.WORKFLOW_SENSOR_CLEANED):
+      return {...state, workflowAction: {
+          ...state.workflowAction, workflowData: {
+            ...state.workflowAction.workflowData,
+            sensor: [
+              ...initialState.workflowAction.workflowData.sensor,
+              {property: action.payload.property, value: action.payload.value}
+            ]
+          }
+        }};
+    case (WorkflowsActions.WORKFLOW_ADD_EMPTY_JOB):
+      const emptyJobData = {order: action.payload, job: []};
+      const jobs = [...state.workflowAction.workflowData.jobs, emptyJobData];
+      return {...state, workflowAction: {
+          ...state.workflowAction, workflowData: {
+            ...state.workflowAction.workflowData, jobs: [...jobs]
+          }
+        }};
+    case (WorkflowsActions.WORKFLOW_JOB_CHANGED):
+      const oldJobDataOption = state.workflowAction.workflowData.jobs.find(job => job.order == action.payload.order);
+      const oldJobData: {property: string, value: any}[] = !!oldJobDataOption ? oldJobDataOption.job : [];
+      const filteredOldJobData = oldJobData.filter(jobEntry => jobEntry.property !== action.payload.jobEntry.property);
+      const updatedJobData = [
+        ...filteredOldJobData,
+        {property: action.payload.jobEntry.property, value: action.payload.jobEntry.value}
+      ];
+      const updatedJobsData = [
+        ...state.workflowAction.workflowData.jobs.filter(jobEntries => jobEntries.order !== action.payload.order),
+        {order: action.payload.order, job: updatedJobData}
+      ];
+      return {...state, workflowAction: {
+          ...state.workflowAction, workflowData: {
+            ...state.workflowAction.workflowData, jobs: [...initialState.workflowAction.workflowData.jobs, ...updatedJobsData]
+          }
+        }};
+    case (WorkflowsActions.WORKFLOW_JOB_CLEANED):
```

Comment:
I think I would rename this to `WORKFLOW_JOB_TYPE_SWITCHED`. Imo, it's not really a clean because you still need to keep the value of the job type. Moreover, cleaning even the job name when switching the job type is not expected behavior imo, but this can be addressed in the future

Reply:
The same argument doesn't exactly hold for `WORKFLOW_SENSOR_CLEANED`, but for the sake of consistency I would change it as well.

Reply:
I've changed that. I think job cleaning is fine, I did it intentionally. But we can discuss it during next planing session.